src/pocketmine/block/Block.php

```diff
@@ -279,6 +279,37 @@ public static function init(){
 			}
 		}
 	}
+	
+	/**
+	 * @param int $blockID
+	 *
+	 * @return int 0-15
+	*/
+	public static function getSkyLightResistance($blockID){
+		#$block = new self::$list[$blockID]();
+		if(self::$transparent[$blockID]){
+			if($blockID == self::WATER || $blockID == self::ICE){
+				return 3;
+			}
+			return 0;
+		}
+		if(self::$solid[$blockID]){
+			return 15;
+		}
+		return -1;
+	}
+	
+	
+	/**
+	 * @param int $blockID
+	 *
+	 * This function must always return values in range of 1-15, otherwise it could lead to a freeze.
+	 *
+	 * @return int 1-15
+	*/
+	public static function getHorizontalSkyLightResistance($blockID){
```

Comment:
horizontal spread skylight light is always decreasing by one. This is the fastest and cleanest way to implement this probably...